apps/omg_eth/lib/omg_eth/root_chain/hack_naming.ex

```diff
@@ -0,0 +1,102 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Eth.HackNaming do
+  @moduledoc """
+  Adapt to naming from contracts to elixir-omg.
+
+  I need to do this even though I'm bleeding out of my eyes.
+  """
+  def deposit(data) do
+    # key is naming coming from plasma contracts
+    # value is what we use
+    contracts_naming = [{"token", :currency}, {"depositor", :owner}, {"blknum", :blknum}, {"amount", :amount}]
+
+    reduce_naming(data, contracts_naming)
+  end
+
+  # we always call it output_index, which is kinda weird?
```

Comment:
not that much. An `input` is also an `output` :)

Reply:
ah OK, yeah, what adds to the confusion here is that it used to be always `output_index`, when there was a single event and inputs were 0->3 and outputs 4->7. So agreed on weird, this is a missed leftover after the hasty push to ALD compatibility :(.

When the overall way you want to handle this renaming crystalizes, ping me to help out on how to reduce the eyebleed in this case.

The goal at the end of the pipeline is to have `{:input | :output, index}` and there is `OMG.EthereumEventListener.Preprocessor` that already does part of this transformation. It could also translate the specific `"outputIndex" | "inputIndex"` into just `index`.

Reply:
yeah awesome! I think we can give this another pass once things settles a little bit!